ui/page/dcrdex/dex_onboarding_page.go

```diff
@@ -984,58 +1012,99 @@ func (pg *DEXOnboarding) HandleUserInteractions() {
 				return
 			}
 
-			// Initialize with password now, if dex password has not been
-			// initialized.
 			pg.isLoading = true
-			if !dexc.InitializedWithPassword() {
+			if dexc.InitializedWithPassword() {
 				go func() {
-					// Set password.
-					pg.dexPass = []byte(pg.passwordEditor.Editor.Text())
-					if err := dexc.InitWithPassword(pg.dexPass, nil); err != nil {
-						pg.isLoading = false
-						pg.notifyError(err.Error())
-						return
-					}
-
-					// Login.
-					err := dexc.Login(pg.dexPass)
-					if err != nil {
-						pg.isLoading = false
-						pg.notifyError(err.Error())
-						return
-					}
-
 					pg.postBond()
+					pg.isLoading = false
 				}()
 
+				break
+			}
+
+			// DEX has not been initialized with a password, do it now.
+			go func() {
+				// Set password.
+				pg.dexPass = []byte(pg.passwordEditor.Editor.Text())
+				if err := dexc.InitWithPassword(pg.dexPass, nil); err != nil {
+					pg.isLoading = false
+					pg.notifyError(err.Error())
+					return
+				}
+
+				// Login.
+				err := dexc.Login(pg.dexPass)
+				if err != nil {
+					pg.isLoading = false
+					pg.notifyError(err.Error())
+					return
+				}
+
+				pg.postBond()
+			}()
```

Comment:
The if statement above can be placed inside this goroutine, no need for 2 goroutines. They both eventually call `pg.postBond()`, so start only one goroutine, inside it, check if you need to call dexc.Init before calling pg.postBond.